### PR TITLE
CentOS 7 packaging

### PR DESCRIPTION
**Pull request by gvde**
Our spec file and additional conf files to create a rpm for CentOS 7.

### CODE REVIEW

**Comment by jornane** [comment]
I've worked on a [radsecproxy.spec](https://git.sr.ht/~jornane/radsecproxy.spec/tree/master/radsecproxy.spec) file myself, also taking some inspiration from this PR. It should work on Fedora, CentOS and RHEL. I still need to do some testing. It'd be nice to have this in EPEL, but for now I've built it on [COPR](https://copr.fedorainfracloud.org/coprs/jornane/radsecproxy/package/radsecproxy/).

File: packaging/rpm/SPECS/radsecproxy.spec [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+mkdir -p %{buildroot}%{_prefix}/lib/tmpfiles.d
+cp radsecproxy.tmpfiles %{buildroot}%{_prefix}/lib/tmpfiles.d/%{name}.conf
+#mkdir -p %{buildroot}%{_localstatedir}/log/%{name}
+mkdir -p %{buildroot}%{_localstatedir}/log
+touch %{buildroot}%{_localstatedir}/log/%{name}.log
+mkdir -p %{buildroot}%{_localstatedir}/run/%{name}
+
+%post
+semanage fcontext -a -t radiusd_exec_t '%{_sbindir}/radsecproxy' 2>/dev/null || :
+restorecon %{_sbindir}/radsecproxy || :
```
Comment (jornane) [inline packaging/rpm/SPECS/radsecproxy.spec]: Is this necessary? Marking the `radsecproxy` binary as a binary belonging to `radiusd` (FreeRADIUS)?

File: packaging/rpm/SPECS/radsecproxy.spec [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+#semanage fcontext -a -t radiusd_log_t '%{_localstatedir}/log/%{name}(/[^/]*)?' 2>/dev/null || :
+#test -d %{_localstatedir}/log/%{name} && restorecon -R %{_localstatedir}/log/%{name} || :
+semanage fcontext -a -t radiusd_unit_file_t '%{_unitdir}/%{name}.service' 2>/dev/null || :
+restorecon %{_unitdir}/%{name}.service || :
```
Comment (jornane) [inline packaging/rpm/SPECS/radsecproxy.spec]: Is this necessary? The default rules already mark everything in `%{_unitdir}` as `systemd_unit_file_t`. Why should it be marked as `radiusd_unit_file_t` instead?

File: packaging/rpm/SPECS/radsecproxy.spec [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+systemctl daemon-reload
+systemctl try-restart %{name}.service
+systemctl try-restart rsyslog.service
```
Comment (jornane) [inline packaging/rpm/SPECS/radsecproxy.spec]: I noticed that in your earlier commits, you included some `rsyslog`-specific things in your configuration, but you removed those later on. I guess restarting `rsyslog.service` is not needed anymore either?

File: packaging/rpm/SPECS/radsecproxy.spec (hunk 1 above)
Comment (gvde) [reply packaging/rpm/SPECS/radsecproxy.spec]: Well, I kind of used that as "shortcut" to get access to all the necessary ports as radsecproxy.

File: packaging/rpm/SPECS/radsecproxy.spec (hunk 2 above)
Comment (gvde) [reply packaging/rpm/SPECS/radsecproxy.spec]: I don't remember exactly why I have put it there but I guess I followed the spec file of radiusd.

File: packaging/rpm/SPECS/radsecproxy.spec (hunk 3 above)
Comment (gvde) [reply packaging/rpm/SPECS/radsecproxy.spec]: You are right. We use rsyslog for the logging on our servers but I then removed that to match the original radsecproxy.conf settings.

**Comment by jornane** [comment]
Thanks @gvde! I want to propose some changes to make it work on Fedora and RHEL but I can't make changes to this PR, but I can make a new one #57. Let me know what you think!

**Comment by robert-scheck** [comment]
I would like to encourage you to contribute to the regular radsecproxy RPM package in Fedora and EPEL rather spinning up own COPR repositories.

**Comment by jornane** [comment]
@robert-scheck already made a .spec file that is already accepted in Fedora

https://src.fedoraproject.org/rpms/radsecproxy/tree/master

**Comment by fmauchle** [comment]
Referring to official package instead.

**Pull request closed without merging**